### DISCUSSION TITLE
question about kernel name

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#4551] [General]
When the kernel name is "distance" (i.e. <class distance>), the compiler error implies that the name is in conflict with something defined in sycl_span.hpp. 

The compiler reported the error with the HIP/ROCm backend. The compiler does not report the error for the CUDA backend. Could you reproduce the error using any sycl program ?

```
error: invalid operands to binary expression ('size_t' (aka 'unsigned long') and 'distance')
    _SYCL_SPAN_ASSERT(_Extent == distance(__f, __l),
```

**Comment by AerialMantis** [comment]
This looks to be a conflict between then type declared for the kernel name and the `distance` function calls, from looking at this I'm not sure why this would fail for the HIP backend but not the CUDA backend, though I think the issue is that the calls to `distance` are not namespace qualified, however, I suspect this may be in order to allow for ADL, if that's the case it might be difficult to avoid this.

@bader what do you think?

**Reply by zjin-lcf** [reply]
This may be unavoidable without qualified namespace. I would use another name. Thanks.

**Reply by bader** [reply]
This looks like a bug in sycl_span.hpp.

**Reply by bader** [reply]
https://github.com/intel/llvm/pull/4570 should fix this issue.
